File: proposal/test1.cpp
#include <stdio.h>
using namespace std;
class Person {
public:
    void quack() {
        printf("I Say 'Quack'\n");
    }
    void walk() {
        printf("Person Walking\n");
    }
};
class Duck {
public:
    void quack() {
        printf("Quack!\n");
    }
    void walk() {
        printf("Duck Walk\n");
    }
    void feathers() {
        printf("Ducks have feathers\n");
    }
};
template <class D>
void InTheForest(D* duck) {
    duck->quack();
    duck->walk();
}
int main(int argc, char** argv) {
    auto duck = new Duck();
    auto person = new Person();

    InTheForest(duck);
    InTheForest(person);
}

File: proposal/test1.py
class Person:
    def quack(self):
        print("I say 'Quack'")
    def walk(self):
        print("Person walking")

class Duck:
    def quack(self):
        print("Quack!")
    def walk(self):
        print("Duck walk")
    def feathers(self):
        print("Ducks have feathers")

def InTheForest(duck):
    duck.quack()
    duck.walk()

duck = Duck()
person = Person()

InTheForest(duck)
InTheForest(person)
